// ng2-simple-timer/lib/simple-timer.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var rxjs_1 = require("rxjs");
var angular2_uuid_1 = require("angular2-uuid");
var SimpleTimer = (function () {
    function SimpleTimer() {
        this.timers = {};
        this.subscription = {};
    }
    SimpleTimer.prototype.getTimer = function () {
        return Object.keys(this.timers);
    };
    SimpleTimer.prototype.getSubscription = function () {
        return Object.keys(this.subscription);
    };
    SimpleTimer.prototype.newTimer = function (name, sec, delay) {
        if (delay === void 0) { delay = false; }
        if (name === undefined || sec === undefined || this.timers[name]) {
            return false;
        }
        var t;
        if (delay) {
            t = rxjs_1.timer(sec * 1000, sec * 1000);
        }
        else {
            t = rxjs_1.timer(0, sec * 1000);
        }
        this.timers[name] = { second: sec, t: t };
        return true;
    };
    SimpleTimer.prototype.delTimer = function (name) {
        var _this = this;
        if (name === undefined || !this.timers[name]) {
            return false;
        }
        var s = this.getSubscription();
        s.forEach(function (i) {
            if (_this.subscription[i].name === name) {
                _this.unsubscribe(i);
            }
        });
        delete this.timers[name].t;
        delete this.timers[name];
    };
    SimpleTimer.prototype.subscribe = function (name, callback) {
        if (!this.timers[name]) {
            return '';
        }
        var id = name + '-' + angular2_uuid_1.UUID.UUID();
        this.subscription[id] = {
            name: name,
            subscription: this.timers[name].t.subscribe(callback)
        };
        return id;
    };
    SimpleTimer.prototype.unsubscribe = function (id) {
        if (!id || !this.subscription[id]) {
            return false;
        }
        this.subscription[id].subscription.unsubscribe();
        delete this.subscription[id];
    };
    SimpleTimer = __decorate([
        core_1.Injectable({ providedIn: 'root' })
    ], SimpleTimer);
    return SimpleTimer;
}());
exports.SimpleTimer = SimpleTimer;
